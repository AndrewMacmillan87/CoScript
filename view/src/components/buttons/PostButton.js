import React from "react";
import PropTypes from "prop-types";

import "./buttons.css";

const PostButton = ({ onClick, classes, value }) => {
    return (
        <input
            onClick={onClick}
            className={classes}
            type='submit'
            value={value}
        />
    );
};

PostButton.defaultProps = {
    value: "Verify"
};

PostButton.propTypes = {
    onClick: PropTypes.func,
    classes: PropTypes.string,
    value: PropTypes.string
};

export default PostButton;
