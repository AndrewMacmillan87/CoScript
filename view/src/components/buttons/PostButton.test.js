import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Provider } from "react-redux";
import { reduxStore } from "../../redux/store/reduxStore";

import PostButton from "./PostButton";

describe("PostButton component", () => {
    test("renders without crashing", () => {
        global.scrollTo = jest.fn();
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={reduxStore}>
                <Router>
                    <PostButton />
                </Router>
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        global.scrollTo = jest.fn();
        const tree = renderer
            .create(
                <Provider store={reduxStore}>
                    <Router>
                        <PostButton />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render a single input element with a className that matches the 'classes' prop", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <PostButton
                        onClick={() => {}}
                        classes='test-class'
                        value='Test'
                    />
                </Router>
            </Provider>
        );
        expect(wrapper.find("input").length).toEqual(1);
        expect(wrapper.find({ className: "test-class" }).length).toEqual(1);
    });
});
