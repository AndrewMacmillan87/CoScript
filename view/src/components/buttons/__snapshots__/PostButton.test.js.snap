// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostButton component renders correctly 1`] = `
<input
  type="submit"
  value="Verify"
/>
`;
